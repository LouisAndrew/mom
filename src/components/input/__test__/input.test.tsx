import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Icon } from '@iconify/react';
import accessibleIcon from '@iconify/icons-uil/accessible-icon-alt';

import Input from '..';

describe('Input element', () => {
    const mockHandleChange = jest.fn(() => {});
    const mockPlaceholderText: string = 'input here';

    afterEach(cleanup);
    const el: React.ReactElement = (
        <Input
            placeholderText={mockPlaceholderText}
            handleChange={mockHandleChange}
        />
    );
    const elWithIcon: React.ReactElement = (
        <Input
            placeholderText={mockPlaceholderText}
            handleChange={mockHandleChange}
            icon={<Icon icon={accessibleIcon} data-testid="icon" />}
        />
    );

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('should render with placeholder text correctly', () => {
        const { queryByPlaceholderText } = render(el);
        const inputEl: HTMLElement | null = queryByPlaceholderText(
            mockPlaceholderText
        );

        if (inputEl) {
            expect(true).toBe(true);
        } else fail();
    });

    it('should react to user changes', () => {
        const { getByPlaceholderText } = render(el);
        const inputEl: HTMLElement = getByPlaceholderText(mockPlaceholderText);

        fireEvent.change(inputEl, { target: { value: 'abcd' } });
        expect(mockHandleChange).toBeCalled();
    });

    it('should render when given a certain icon correctly', () => {
        const { queryByTestId } = render(elWithIcon);
        const icon: HTMLElement | null = queryByTestId('icon');

        if (icon) {
            expect(true).toBe(true);
        } else fail();
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
