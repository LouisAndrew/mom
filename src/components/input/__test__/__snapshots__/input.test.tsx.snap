// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input element matches snapshot 1`] = `
<input
  className="sc-AxgMl kxfDk"
  fontFamily="body"
  fontSize={
    Array [
      1,
      1,
      2,
      2,
    ]
  }
  fontWeight="bold"
  id="input"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          SyntheticEvent {
            "_dispatchInstances": null,
            "_dispatchListeners": null,
            "_targetInst": null,
            "bubbles": null,
            "cancelable": null,
            "currentTarget": [Function],
            "defaultPrevented": null,
            "dispatchConfig": null,
            "eventPhase": null,
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": null,
            "nativeEvent": null,
            "target": null,
            "timeStamp": [Function],
            "type": null,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="input here"
/>
`;
